components/Navbar.tsx

```diff
@@ -0,0 +1,195 @@
+          <Link href="/" passHref>
+            <a
+              className={classNames(
+                { "xl:pl-11": !authPageOnly },
+                "inline-flex items-center cursor-pointer "
+              )}
+            >
+              <Image src={Logo} width="160" height="36" alt="logo" />
+            </a>
+          </Link>
+          <ul className="items-center hidden space-x-8 lg:flex">
+            {navBarLink.map((data, index) => {
+              return <NavLink key={index} title={data.title} url={data.url} />;
+            })}
+          </ul>
+          <ul className="items-center hidden lg:flex">
+            <li>
+              <Link href="/sign-in">
+                <a>
+                  <Button outline className="uppercase">
+                    Sign in
+                  </Button>
+                </a>
+              </Link>
+            </li>
+            <li>
+              <Link href="/sign-up">
+                <a>
+                  <Button className="ml-4 uppercase">Try it now</Button>
+                </a>
+              </Link>
+            </li>
+          </ul>
+
+          <div className="lg:hidden ">
+            <button
+              aria-label="Open Menu"
+              title="Open Menu"
+              className="p-2  -mr-1 transition duration-200 rounded focus:outline-none focus:shadow-outline "
+              onClick={() => setIsMenuOpen(true)}
+            >
+              <Image
+                src={HamburgerMenu}
+                width="30px"
+                height="30px"
+                alt="hamburger icon"
+              />
+            </button>
+            {setIsMenuOpen && (
+              <Menu isMenuOpen={isMenuOpen}>
+                <>
+                  <div className="relative">
+                    <div className={style.logo}>
+                      <Image
+                        src={WhiteLogo}
+                        width="160"
+                        height="36"
+                        alt="icon logo"
+                      />
+                    </div>
+                  </div>
+                  <button
+                    aria-label="Close"
+                    className="absolute top-3 right-5 text-5xl text-white cursor-pointer "
+                    onClick={() => setIsMenuOpen()}
+                  >
+                    &times;
+                  </button>
+                  <MenuContainer>
+                    {navBarLink.map((data, index) => {
+                      return (
+                        <MenuItem
+                          key={index}
+                          title={data.title}
+                          url={data.url}
+                        />
+                      );
+                    })}
+                    <MenuItem url="/sign-up" title="Sign Up" />
+                  </MenuContainer>
+                </>
+              </Menu>
+            )}
+          </div>
+        </div>
+      </div>
+    </header>
+  );
+};
+
+Navbar.propTypes = {
```

Comment:
Since we are using Typescript, this is no longer required as it can be defined as type.